packages/salesforcedx-vscode-core/src/commands/DeployExecutor.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {
+  getRelativeProjectPath,
+  Row,
+  Table
+} from '@salesforce/salesforcedx-utils-vscode';
+import { ComponentSet, DeployResult } from '@salesforce/source-deploy-retrieve';
+import { RequestStatus } from '@salesforce/source-deploy-retrieve/lib/src/client/types';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { WorkspaceContext } from '../context/workspaceContext';
+import { handleDeployDiagnostics } from '../diagnostics';
+import { nls } from '../messages';
+import { SourceTrackingService } from '../services';
+import { DeployQueue } from '../settings';
+import { SfdxPackageDirectories } from '../sfdxProject';
+import { BaseDeployExecutor } from './baseDeployCommand';
+import { DeployRetrieveExecutor } from './baseDeployRetrieve';
+
+export abstract class DeployExecutor<T> extends DeployRetrieveExecutor<T> {
```

Comment:
Extracted to make unit testing easier

Reply:
In the current PR I left this in place in baseDepRetr.  I think the PR is cleaner this way, but let me know if you think I should pull these out again and put them under a executors dir.